src/main/java/org/spongepowered/mod/service/scheduler/AsyncScheduler.java

```diff
@@ -0,0 +1,743 @@
+ * is 50ms, but due to overhead and latency, the monotonous timing of Tick based Tasks is not guaranteed.</p>
+ *
+ * <p> {@link AsyncScheduler} implements {@link AsynchronousScheduler} interface and uses milliseconds as the time
+ * unit.  Tasks created with parameters (if any) that involve delays and periods based on milliseconds and relative
+ * to the wall-clock of the host system.  Caveat: the wall-clock time is used for measuring elapsed time, but the
+ * actual date/time of the host system or changes to the host date/time will not affect the scheduling of Tasks.
+ * Because the AsyncScheduler is running its own thread all tasks that  execute from this scheduler are not
+ * synchronous with the game data.  In other-words, the Tasks that are executed by the AsyncScheduler are not
+ * thread-safe with the game data.  Plugin authors should use care when leveraging concurrency in their plugins
+ * when those plugins access game data as a result of a Asynchronous {@link Task} running.  Several resources can
+ * help the plugin author:  The Java Concurrency book by Lea and Effective Java by Bloch for information handling
+ * concurrency issues in their plugins.</p>
+ * *
+ * <p>Tasks can be created using the API in the {@link AsynchronousScheduler} interface. The access to this Scheduler
+ * through the {@link org.spongepowered.api.Game} interface method AsynchronousScheduler getAsyncScheduler();
+ * Plugin authors never cause the creation of a scheduler, the scheduler (Asynchronous and {@link SyncScheduler}
+ * are each Singletons.</p>
+ *
+ *
+ */
+public class AsyncScheduler implements AsynchronousScheduler {
+
+    // Simple Moore State machine state. (http://en.wikipedia.org/wiki/Moore_machine)
+    private MachineState sm;
+    // The simple private queue of all pending (and running) ScheduledTasks
+    private final Queue<ScheduledTask> taskList = new ConcurrentLinkedQueue<ScheduledTask>();
+    // Adjustable timeout for pending Tasks
+    private long minimumTimeout = Long.MAX_VALUE;
+    private long lastProcessingTimestamp;
+    // Locking mechanism
+    private final Lock lock = new ReentrantLock();
+    private final Condition condition = lock.newCondition();
+    // The dynamic thread pooling executor of asynchronous tasks.
+    private static ExecutorService executor;
+
+    private AsyncScheduler() {
+        new Thread(new Runnable() {
+            public void run() {
+                SMachine();
+            }
+        }).start();
+    }
+
+    /**
+     * <p>Unlike the State of a ScheduledTask, the @MachineState of this Asynchronous Scheduler is private and not an
+     * enumeration of values that have any user facing purpose.  This is why the enumeration is
+     * defined within the class and inaccessible to user code.</p>
+     */
+    private enum MachineState {
+        PRE_INIT,
+        INIT,
+        RUN,
+        NOT_RUNNING,
+    }
+
+    // The Moore State Machine for this Scheduler
+    private void SMachine() {
```

Comment:
Pre_Init ?

I would not want to go that way on pure enum integral values.  Do we have to ?